doc/use_cli_effectively.md

```diff
@@ -0,0 +1,88 @@
+
+       Less known, but CLI also has built-in support on processing "--ids" in parallel to achieve the same effect of xargs
+
+          az vm list -d -g my_rg --query "[?powerState=='VM stopped'].id" -o tsv | tr '\r\n' ' ' | { read ids ; az vm start --ids $ids ; }
+
+## Async executions ##
+  This becomes handy for a few scenarios:
+  1. Clean up
+
+          az group delete -n my_rg --no-wait
+
+  2. For performance, don't wait on each operation, rather wait for all at the end
+      
+          az vm create -g my_rg -n vm1 --image centos --no-wait
+          az vm create -g my_rg -n vm2 --image centos --no-wait
+           
+          az vm wait --created --ids $(az vm list -g my_rg --query "[].id" -o tsv)
+
+## Generic update command ##
+  The hallmark is the `"--add", "--set", "--remove"` generic arguments. Being generic means it can do more things but at the cost of usabilities. For common scenarios, CLI always provides custom commands to make your life easy, but leave the rest for you to explore through the generic updater. A few notes here to ease the pain:
+
+  1. You will need some patience to pull it through.
+  2. Do check out whether or not the update command has the convinient arguments exposed for your scenario. If yes, use those.
+  3. The property path used with the generic arguments are tied to the output of "show" command. For example, before you try out "az vm update", do invoke the "az vm show" command, read the output, and figure out the right path.
+  4. Checking out working examples is the best way to get started. `az vm update -h` has the good ones.
+  5. Do remember that generic arguments of `--set` and `--add` takes a list of key value pairs in the format of `<key1>=<value1> <key2>=<value2>`. You will need them to construct non trivial payload. If the input gets complex, using json string will be the best bet, e.g. to attach a new disk:
+
+         az vm update -g my_rg -n my_vm --add storageProfile.dataDisks "{\"createOption\": \"Attach\", \"managedDisk\": {\"id\": \"/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/yg/providers/Microsoft.Compute/disks/yg-disk\"}, \"lun\": 1}"
+
+  6. Of course, leverage CLI's `@<file>` convention, you can put the json string to a file and simplify the command
+
+         az vm update -g my_rg -n my_vm --add storageProfile.dataDisks @~/my_disk.json
+
+## Single quote vs double quote ##
+  This becomes a topic because when the command shell(bash, zsh, windows command prompt, etc) parses the CLI command, it will interpret the quotes. To avoid the surprise, a few suggestions here:
+  1. If the argument value contains whitespace, you have to use quotes to wrap them.
+  2. In Bash, both single and double quotes will be intepreted; while in windows command, only double quotes get handled which means single quotes will be part of values. So be careful on this difference.
```

Comment:
Maybe "interpreted as string" or "handled as strings", what do you think?

Reply:
bump.

Reply:
i have fixed this